pkg/iosrc/iosrc.go

```diff
@@ -0,0 +1,134 @@
+//go:generate mockgen -destination=./mock/mock_source.go -package=mock github.com/brimsec/zq/pkg/iosrc Source
+
+package iosrc
+
+import (
+	"errors"
+	"io"
+	"os"
+	"sync"
+	"time"
+)
+
+const FileScheme = "file"
+
+var DefaultRegistry = &Registry{
```

Comment:
I suggested to @mattnibs off-PR that in a potential follow-up PR, we could remove the ability to add Sources to registries at runtime, he said he'd investigate.